source/air/src/typecheck.rs

```diff
@@ -141,6 +141,13 @@ fn check_bv_unary_exprs(
             }
             Ok(Arc::new(TypX::BitVec(w_new)))
         }
```

Comment:
It seems like `check_bv_unary_exprs` should in general return an `Err` when the exprs have the wrong type, rather than panicking.  Also, it doesn't look like you need multiple exprs here, since you only use `exprs[0]`.

Reply:
Thanks for the comments :)